File: Advance learning/DSA/1st Sem/MovieTime.py
from collections import deque

def main():
    movies = deque()
    snacks = deque()


    for i in range(1, 4):
        movie = input(f"Enter movie {i} of 3: ")
        movies.append(movie)
    

    for i in range(1, 4):
        snack = input(f"Enter snack {i} of 3: ")
        snacks.append(snack)


    print("\nMovies to watch are:", list(movies))
    print("Snacks available are:", list(snacks))
    print()

    while snacks:
        select = input("Press S when you finish a snack: ").strip().lower()
        if select == 's':
            snacks.popleft() 
            if not snacks:
                print("No no more snacks exist na")
            else:
                print("Remaining snacks:", list(snacks))
        else:
            print("Invalid input")



main()

File: Advance learning/DSA/1st Sem/MovieTime.java
import java.util.LinkedList;
import java.util.Queue;
import java.util.*;
public class MovieTime {
    public static void main(String[] args){
        Scanner s = new Scanner(System.in);
        Queue<String> movies = new LinkedList<>();
        Queue<String> snacks = new LinkedList<>();
        String movie;
        String snack;
        while(true){
            
        for(int i = 1; i <= 3; i++){
            System.out.println("Enter 3 movie he/she wants to watch: movies " + i + " of 3");
            movie = s.nextLine();
            //System.out.println(movie);
            movies.offer(movie);
        }
        for(int i = 1; i <= 3; i++){
            System.out.println("Enter snack " + i + " of 3");
            snack = s.nextLine();
            //System.out.println(snack);
            snacks.offer(snack);
        }
        break;
        }
        
        System.out.println();
        System.out.println("Movies to watch are: " + movies);
        System.out.println("Snacks available are: " + snacks);
        System.out.println();
        
        while(!snacks.isEmpty()){
            System.out.println("Press S to each time you finish a snack ");
            char select = s.next().charAt(0);
            if(select == 'S' || select == 's'){
                snacks.poll();
                if(snacks.isEmpty()){
                    System.out.println("No no snacks exist na");
                } else {
                    System.out.println("Remaining: " + snacks);
                }
            } else {
                System.out.println("Invalid Input");
            }
        }
    }
}